src/wilson/dom.cljs

```diff
@@ -174,9 +174,15 @@
   (update elem 1 merge-attrs attrs))
 
 (defn with-class
-  "Adds a class to an element."
-  [cls elem]
-  (with-attrs {:class cls} elem))
+  "Adds a class to a component."
+  [cls component]
+  (if (vector? component)
```

Comment:
Yeah, I dislike having two drastically different ways you use something even if it has to do something drastically different on the inside